### PR TITLE
Data types for Go

### LINKED ISSUE #91: Add Go data types
# Description

Data Types for Go aren't presently on Code Thesaurus. This issue is to add that concept to the system.

# Requirements

* Copy `web/thesauruses/_meta/data_types.json` to the right Go directory in `web/thesauruses/`
* Edit the data_types.json file to accurately reflect the latest major version of Go
* Delete the `"name":` fields before your PR
* Add in `"comment":` if you wish to explain something in the code
* Add in `"not-implemented": "true"` if the concept doesn't exist in the language. You can then leave `"code":` blank.

# Claiming This Issue

Please leave a comment below so I can assign you to it. This is to prevent multiple people from working on the same issue. 

Check out [CONTRIBUTING.md](https://github.com/codethesaurus/codethesaur.us/blob/main/CONTRIBUTING.md) for more information on how to contribute. 

Check out [https://docs.codethesaur.us/thesaurus/](https://docs.codethesaur.us/thesaurus/) for information on how Code Thesaurus works and how to add to the thesauruses.

# Need Help?

Leave a comment on here with your question, or feel free to ask [@geekygirlsarah](https://twitter.com/geekygirlsarah) on Twitter. You can also email coreteam@codethesaur.us too.

# Thanks!

We really value and appreciate the additions people make to Code Thesaurus, no matter how large or small. So thank you for taking the time to work on this!

### PR DESCRIPTION
**Pull request by carlqt**
## 📝 What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## 📖 Description
Fix for https://github.com/codethesaurus/codethesaur.us/issues/91

Adds Data Type docs for `Go`

## ✅ QA Instructions, Screenshots

image

### CODE REVIEW

**Comment by geekygirlsarah** [comment]
I'm curious: you have no code in here. Is there no actual keywords or anything to refer to these types?

**Comment by carlqt** [comment]
Hi @geekygirlsarah , just curious what's the expectations here because I just pretty much followed the pattern that I see in the codebase. Is there something you have in mind so I could adjust my implementation?

**Comment by geekygirlsarah** [comment]
If you look at C#'s for example: https://codethesaur.us/reference/?concept=data_types&lang=csharp and https://github.com/codethesaurus/codethesaur.us/blob/main/web/thesauruses/csharp/data_types.json

It lists the keywords that are those data types in the `code` keys. While you have the `code` blocks,they all looked empty. You did do the `not-implemented` parts right.

So I'm wondering: are there data type keywords in Go? (I don't know it well enough to know and I'm on a plane and not in a good spot to verify right now.)

**Comment by carlqt** [comment]
Ahhh now I got it. Yes, there are. Let me update my implementations then. Thanks for the review

**Comment by carlqt** [comment]
Updated the PR

File: web/thesauruses/go/data_types.json [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    "signed_float_as_object": {
+      "not-implemented": "true",
+      "code": ""
+    },
+    "unsigned_float_as_object": {
+      "not-implemented": "true",
+      "code": ""
+    },
+    "character": {
+      "code": ""
```
Comment (geekygirlsarah) [inline web/thesauruses/go/data_types.json]: Is this also a `not-implemented`?

File: web/thesauruses/go/data_types.json [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    },
+    "string_as_object": {
+      "code": "string",
+      "comment": "string is a read only slice of bytes in golang"
+    },
+    "string_as_array": {
+      "code": ""
```
Comment (geekygirlsarah) [inline web/thesauruses/go/data_types.json]: Same for these below, are they not-implemented?

File: web/thesauruses/go/data_types.json [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    "unsigned_integer_64_bit": {
+      "code": "uint64"
+    },
+    "signed_integer_as_object": {
+      "not-implemented": "true",
+      "code": ""
+    },
+    "unsigned_integer_as_object": {
+      "not-implemented": "true",
+      "code": ""
```
Comment (geekygirlsarah) [inline web/thesauruses/go/data_types.json]: For the `not-implemented` ones, you can remove the `code` lines too.

**Review by geekygirlsarah — CHANGES_REQUESTED** [review]

**Comment by carlqt** [comment]
Thanks for the review. Question, what if there are types here that I'm unsure like `complex_as_object`? What should I do for that?

**Comment by geekygirlsarah** [comment]
> 
> 
> Thanks for the review. Question, what if there are types here that I'm unsure like `complex_as_object`? What should I do for that?

Good question! I tried to make descriptions in the _meta files for what they are. If they're not clear, you can always ask here or email/tweet too. (For example, that's a native object that can store a complex number (like `3 + 4i`).)

You could also leave them essentially undefined (like delete the concept ID `complex_as_object` all over) and that'll show that its state is unknown. I don't prefer this, but the site is set up to accept this as a potential thing as the meta files can evolve when these files may not always be in sync with them.

**Comment by carlqt** [comment]
> You could also leave them essentially undefined (like delete the concept ID complex_as_object all over) and that'll show that its state is unknown. I don't prefer this, but the site is set up to accept this as a potential thing as the meta files can evolve when these files may not always be in sync with them.

You mentioend that you don't prefer leaving it `blank` if the type is unknown, so what's the preference? I've updated the PR with the new changes maybe you can check if that's satisfactory or needs more tweak.

**Comment by geekygirlsarah** [comment]
You can use `"not-implemented": true` on it instead of code. (You can optionally still have a comment)

**Comment by carlqt** [comment]
Hi, I've updated the PR

**Comment by geekygirlsarah** [comment]
You force pushed? It makes it hard to see what changed since I asked for revisions! But reviewing it all again, it looks right so I'll merge it.

Don't forget to sign up for Hacktoberfest if you haven't so this can count towards the shirt/stickers!

**Comment by carlqt** [comment]
@geekygirlsarah Thanks. Sorry, I forced push out of habit 😭 .

**Comment by geekygirlsarah** [comment]
It's alright, it's a fairly small file and easy to re-review.

**Pull request merged**